python/cudf/cudf/core/dataframe.py

```diff
@@ -4450,6 +4450,97 @@ def skew(
             **kwargs,
         )
 
+    def clip(self, lower=None, upper=None, inplace=False):
+        """
+        Trim values at input threshold(s).
+
+        Assigns values outside boundary to boundary values.
+        Thresholds can be singular values or array like,
+        and in the latter case the clipping is performed
+        element-wise in the specified axis.
+
+        Parameters
+        ----------
+        lower : scalar or array_like, default None
+            Minimum threshold value. All values below this
+            threshold will be set to it. If it is None,
+            there will be no clipping based on lower.
+        upper : scalar or array_like, default None
+            Maximum threshold value. All values below this
+            threshold will be set to it. If it is None,
+            there will be no clipping based on upper.
+        inplace : bool, default False
+
+        Returns
+        -------
+        Clipped DataFrame
+
+        Examples
+        --------
+        >>> import cudf
+        >>> df = cudf.DataFrame({"a":[1, 2, 3, 4], "b":['a', 'b', 'c', 'd']})
+        >>> df.clip(lower=[2, 'b'], upper=[3, 'c'])
+           a  b
+        0  2  b
+        1  2  b
+        2  3  c
+        3  3  c
+
+        >>> df.clip(lower=None, upper=[3, 'c'])
+           a  b
+        0  1  a
+        1  2  b
+        2  3  c
+        3  3  c
+
+        >>> df.clip(lower=[2, 'b'], upper=None)
+           a  b
+        0  2  b
+        1  2  b
+        2  3  c
+        3  4  d
+
+        >>> df = cudf.DataFrame({"a":[1, 2, 3, 4], "b":[2, 3, 4, 5]})
+        >>> df.clip(lower=2, upper=4)
+           a  b
+        0  2  2
+        1  2  3
+        2  3  4
+        3  4  4
+
+        >>> df.clip(lower=2, upper=3, inplace=True)
+        >>> df
+           a  b
+        0  2  2
+        1  2  3
+        2  3  3
+        3  3  3
+        """
+
+        if lower is None and upper is None:
+            return self if inplace is True else self.copy(deep=True)
```

Comment:
If `inplace` is True it's expected to modify the object inplace and return None. I think we can change this to:
```suggestion
            return None if inplace is True else self.copy(deep=True)
```